paperqa/types.py

```diff
@@ -650,13 +650,19 @@ def __add__(self, other: DocDetails | int) -> DocDetails:
                 # if we have multiple keys, we wipe them and allow regeneration
                 merged_data[field] = None  # type: ignore[assignment]
 
-            elif (
-                field in {"citation_count", "year", "publication_date"}
-                and self_value is not None
-                and other_value is not None
-            ):
+            elif field in {"citation_count", "year", "publication_date"}:
                 # get the latest data
-                merged_data[field] = max(self_value, other_value)
+                if self_value is None or other_value is None:
```

Comment:
Can you just add a comment explaining the race condition-y thing you encountered that this fixes, either on this line or just above this `if` clause